internal/controller/dnsrecord_controller.go

```diff
@@ -120,8 +120,33 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// Update the logger with appropriate record/zone metadata from the dnsRecord
 	ctx, logger = r.setLogger(ctx, baseLogger, dnsRecord)
 
+	if dnsRecord.Status.ProviderEndpointsRemoved() {
+		logger.V(1).Info("Status ProviderEndpointRemoved is true, finalizer can be removed")
+		logger.Info("Removing Finalizer", "finalizer_name", DNSRecordFinalizer)
+		controllerutil.RemoveFinalizer(dnsRecord, DNSRecordFinalizer)
+		if err = r.Update(ctx, dnsRecord); client.IgnoreNotFound(err) != nil {
+			if apierrors.IsConflict(err) {
+				return ctrl.Result{Requeue: true}, nil
+			}
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
:thinking: I had expected this to be inside the check for a deletion, since we only ever want to remove the finalizer when we are deleting. I guess this is fine, and not really arguing about it, but looks like you could replace [this](https://github.com/Kuadrant/dns-operator/pull/491/files#diff-adacb6a4589c65c2d853c0f59ca02d6b8fb50bb388b41de1d13e99b0d128cfe1R137-R142), with the above no?

Reply:
I was thinking about this from the multi cluster point. If the DNSrecord is on remote, I can see a case where the primary cleans up the DNS entries with out the the remote DNSrecord being marked for deletion. In that case the remote dns-operator would need to remove the finalizer from the resource.  

The spot where you said it could be placed is valid depending on how this is meant to work in multi cluster setup. 

1. If an external entity updates the status to  `ProviderEndpointsRemoved`, the dns-operator should remove the finalizer.
2. If an external entity updates the status to  `ProviderEndpointsRemoved` and the DNSrecord is marked for deletion, the dns-operator should remove the finalizer.

Which of the two statements is the correct behavior?